### PR TITLE
Fix RDBMS registry find method

### LINKED ISSUE #1126: Fix RDBMSRegistry to handle IllegalArgumentException on find()
Based on this https://github.com/spring-cloud/spring-cloud-deployer/issues/176, we need to fix the RDBMS registry find(key) method to throw `IllegalArgumentException` when there is no app found by the key

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Throw IllegalArgumentException when the URI for the given app name is not found

Resolves #1126

### CODE REVIEW

File: spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/RdbmsUriRegistry.java [hunk 1]
```diff
@@ -65,9 +66,9 @@ public URI find(String name) {
 			uriString = jdbcTemplate.queryForObject(SELECT_URI_SQL, String.class, name);
 		}
 		catch (EmptyResultDataAccessException e) {
-			return null;
+			throw new IllegalArgumentException("No URI is registered for ["+ name + "] "+ e);
 		}
-		return uriString != null ? toUri(uriString) : null;
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/RdbmsUriRegistry.java]: I believe `null` or `non-text` values need not be checked here as the `register` operation has validation for those before registering.

Comment (jvalkeal) [inline spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/RdbmsUriRegistry.java]: Well yes, now it does what interface javadoc says.

**Comment by jvalkeal** [comment]
I do get test failures with this
```
Tests in error: 
  testRegisterOverwrite(org.springframework.cloud.dataflow.server.registry.RdbmsUriRegistryTests): No URI is registered for [source.test1] org.springframework.dao.EmptyResultDataAccessException: Incorrect result size: expected 1, actual 0
  testFindAll(org.springframework.cloud.dataflow.server.registry.RdbmsUriRegistryTests): No URI is registered for [source.test1] org.springframework.dao.EmptyResultDataAccessException: Incorrect result size: expected 1, actual 0
  testFind(org.springframework.cloud.dataflow.server.registry.RdbmsUriRegistryTests): No URI is registered for [source.test1] org.springframework.dao.EmptyResultDataAccessException: Incorrect result size: expected 1, actual 0
  testUnregister(org.springframework.cloud.dataflow.server.registry.RdbmsUriRegistryTests): No URI is registered for [source.test1] org.springframework.dao.EmptyResultDataAccessException: Incorrect result size: expected 1, actual 0

Tests run: 229, Failures: 0, Errors: 4, Skipped: 0
```

**Comment by ilayaperumalg** [comment]
@jvalkeal Sorry, I missed it with some changes just before push. will fix it.

**Comment by ilayaperumalg** [comment]
Pushed the changes to handle `IllegalArgumentException` at the `register` method.

As mentioned [here] (https://github.com/spring-cloud/spring-cloud-dataflow/issues/1126#issuecomment-273196002) it might be better to handle all this at the deployer SPI level by returning null on the apps that are not found. But, is that ok to change this now at the deployer level?

**Comment by jvalkeal** [comment]
I think this is ok as I would not just change SPI(maybe later but not in a point release).

**Comment by jvalkeal** [comment]
Merged/squashed per https://github.com/spring-cloud/spring-cloud-dataflow/commit/3148fbf5ced0043e0f68fdd676c900a7b6556a3d

**Pull request closed without merging**